internal/codegen/functions_test.go

```diff
@@ -31,3 +31,26 @@ func Example_quotejoinstrings() {
 	// Output:
 	// "a" "b" "c"
 }
+
+func Example_toYaml() {
```

Comment:
"// This is not a general-purpose YAML parser, and will not parse all valid
// YAML documents."

Which type of documents does it not parse? Would it be helpful to add an example where this will fail?  Perhaps this might help people, though unsure, just brainstorming